### DISCUSSION TITLE
Q: Save html of the initial request

### DISCUSSION THREAD

**Discussion by plesiecki** [GoogleChrome/lighthouse#13827] [Q&A] [answered]
Hello,

I have a quick question. Is it somehow possible to save a html of the initial request? 

Thanks in advance, peace.

**Comment by adamraine** [comment, marked as answer]
We do save the html content for the main document, but this isn't the initial request if there were any redirects.

You should be able to get the html using the node api:
```js
const {artifacts} = await lighthouse(url);
fs.writeFileSync('main.html', artifacts.MainDocumentContent);
```

**Reply by plesiecki** [reply]
Oh, that's lovely. Thanks for an quick answer!
